app/src/main/java/at/aau/serg/network/HttpClient.kt

```diff
@@ -7,9 +7,17 @@ import okhttp3.OkHttpClient
 import okhttp3.Request
 import okhttp3.RequestBody.Companion.toRequestBody
 
-class HttpClient(private var baseUrl: String) {
+class HttpClient private constructor(private val baseUrl: String) {
     private var client: OkHttpClient = OkHttpClient()
 
+    companion object{
+        @Volatile
+        private var INSTANCE: HttpClient? = null
+
+        @Synchronized
+        fun getInstance(baseUrl: String): HttpClient = INSTANCE ?: HttpClient(baseUrl)
```

Comment:
A minor issue here is that we must pass the `baseUrl` whenever we request an instance, even though it is not used after the first call. If we continue to only use `getString(R.string.api_url)` as the `baseUrl` (not sending HTTP requests to other servers for example), we could consider using an `object` for the singleton (does not allow passing parameters) and hard code the `baseUrl`. Opinions @BenjaminHac-Edu @FRALEXAlexander

Reply:
I do not see a case where we need to access other baseUrls.

Reply:
The problem here is that the url can not be easilly hardcoded, because it is stored as a string resource.

Reply:
It could be put in an `init` block, but I will create a separate ticket for this since, as far as I can see, we should be able to do this for other classes as well.

Reply:
I have created #24 as a follow-up where we can take a close look at how we can remove redundant parameters and maybe also inject singletons into other classes.